src/api/middleware/validators/offer.js

```diff
@@ -130,6 +130,160 @@ const create = useExpressValidators([
+        .exists().withMessage(ValidationReasons.REQUIRED).bail()
+        .custom(isObjectId).withMessage(ValidationReasons.OBJECT_ID).bail()
+        .custom(async (_id, { req }) => {
+            try {
+                const offer = await (new OfferService()).getOfferById(_id, req.user);
+                if (!offer) throw new Error(ValidationReasons.OFFER_NOT_FOUND(_id));
+                if (req.user?.company.toString() !== offer.owner.toString()) throw new Error(ValidationReasons.NOT_OFFER_OWNER(_id));
+            } catch (_e) {
+                // Also catches any fail to the DB
+                throw new Error(_e);
+            }
+
+            return true;
+        }),
+
+
+    body("title", ValidationReasons.DEFAULT)
+        .optional()
+        .isString().withMessage(ValidationReasons.STRING)
+        .isLength({ max: OfferConstants.title.max_length }).withMessage(ValidationReasons.TOO_LONG(90))
+        .trim(),
+
+    body("publishDate", ValidationReasons.DEFAULT)
+        .optional()
+        .isISO8601({ strict: true }).withMessage(ValidationReasons.DATE).bail()
+        .custom(async (publishDateRaw, { req }) => {
+            try {
+                const offer = await (new OfferService()).getOfferById(req.body._id, req.user);
+
+                const { publishEndDate: publishEndDateCandidate } = req.body;
+
+                const publishDate = (new Date(Date.parse(publishDateRaw))).toISOString();
+
+                if (publishDate <= (new Date(Date.now())).toISOString()) {
+                    throw new Error(ValidationReasons.MUST_BE_AFTER("currentDate"));
+                }
+
+                // If the new publishEndDate is after the new publishDate, the verification will be done in publishEndDate
+                if (publishDate >= offer.publishEndDate.toISOString() &&
+                        !publishEndDateCandidate) {
+
+                    // end date is earlier than publish date, error!
+                    throw new Error(ValidationReasons.MUST_BE_BEFORE("publishEndDate"));
+                }
+
+            } catch (_e) {
+                // Also catches any fail to the DB
+                throw new Error(_e);
+            }
+            return true;
+        })
+        .toDate(),
+
+    body("publishEndDate", ValidationReasons.DEFAULT)
+        .optional()
+        .isISO8601({ strict: true }).withMessage(ValidationReasons.DATE).bail()
+        .isAfter().withMessage(ValidationReasons.DATE_EXPIRED).bail()
+        .custom(async (publishEndDateCandidate, { req }) => {
+            try {
+                const offer = await (new OfferService()).getOfferById(req.body._id, req.user);
+
+                const { publishDate: publishDateRaw } = req.body;
+                const publishDate = !publishDateRaw
+                    ? publishDateRaw
+                    : (new Date(Date.parse(publishDateRaw))).toISOString();
+                const publishEndDate = (new Date(Date.parse(publishEndDateCandidate))).toISOString();
```

Comment:
You should use the same naming for both cases (avoid *Raw vs *Candidate) -> rename publishDateRaw to publishDateCandidate

Also, same as some other place: The date parsing is not necessary, since you are already validating that the received string is a ISO8601 Date string